File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamHuongViForm.ts
import { BooleanEditor, initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface SanPhamHuongViForm {
    TenHuongVi: StringEditor;
    IsDefault: BooleanEditor;
}

export class SanPhamHuongViForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.SanPhamHuongVi';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SanPhamHuongViForm.init) {
            SanPhamHuongViForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(SanPhamHuongViForm, [
                'TenHuongVi', w0,
                'IsDefault', w1
            ]);
        }
    }
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamHuongVi/SanPhamHuongViForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.SanPhamHuongVi")]
[BasedOnRow(typeof(SanPhamHuongViRow), CheckNames = true)]
public class SanPhamHuongViForm
{
   
    public string TenHuongVi { get; set; }
    public bool IsDefault { get; set; }
}